src/main/java/br/com/alura/comex/controller/AutenticacaoController.java

```diff
@@ -0,0 +1,41 @@
+package br.com.alura.comex.controller;
+
+
+import br.com.alura.comex.config.security.TokenService;
+import br.com.alura.comex.controller.dto.TokenDto;
+import br.com.alura.comex.controller.form.LoginForm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/auth")
+public class AutenticacaoController {
+    @Autowired
+    private AuthenticationManager authManager;
+
+    @Autowired
+    private TokenService tokenService;
+
+    @PostMapping
+    public ResponseEntity<TokenDto> autenticar(@RequestBody @Valid LoginForm form) {
+        UsernamePasswordAuthenticationToken dadosLogin = form.converter();
+
+        try {
+            Authentication authentication = authManager.authenticate(dadosLogin);
+            String token = tokenService.gerarToken(authentication);
+            return ResponseEntity.ok(new TokenDto(token, "Bearer"));
+        } catch (AuthenticationException e) {
+            return ResponseEntity.badRequest().build();
```

Comment:
Uma melhoria que poderia ser feita seria logar a exceção. Poderia ser um `e.printStackTrace()`, mas o ideal seria fazer um log mesmo.

```suggestion
// logar aqui a exception
            return ResponseEntity.badRequest().build();
```

Como fazer logging com Spring: https://www.baeldung.com/spring-boot-logging